### PR TITLE
Tags for Notes

### PR DESCRIPTION
**Pull request by SanjithVenkatesh**
Added crude ability to add tags to notes. Still need to change around the front end as it looks very messy but the back-end code is there.

### CODE REVIEW

File: src/views/NotesPage.vue [hunk 1]
```diff
@@ -98,6 +127,15 @@ export default {
         .catch(error => {
           alert(error);
         });
+    },
+    addTag(newTag) {
+      const vm = this;
+      const tag = {
+        name: newTag,
+        code: newTag.substring(0, 2) + Math.floor(Math.random() * 10000000)
```
Comment (shifuchen98) [inline src/views/NotesPage.vue]: What's the purpose of `code`?

File: src/views/NotesPage.vue [hunk 2]
```diff
@@ -31,6 +43,11 @@
             <h1>{{ note.get("title") }}</h1>
             <p>{{ note.createdAt.toLocaleString("en-US") }}</p>
             <p>{{ note.get("content") }}</p>
+            <h4>Tags</h4>
+            <p>{{ note.get("tags") }}</p>
+            <!-- <ul>
+              <li v-for="{ tag, index } in note.get("tags")" :key="index">{{ tag }}</li>
```
Comment (shifuchen98) [inline src/views/NotesPage.vue]: `{ tag, index }` -> `(tag, index)`

Comment (shifuchen98) [inline src/views/NotesPage.vue]: `{ tag, index }` -> `(tag, index)`

File: src/views/NotesPage.vue (hunk 1 above)
Comment (SanjithVenkatesh) [reply src/views/NotesPage.vue]: I don't know honestly, the vue-multiselect class requires it. If I remove it then the whole thing crashes.

File: src/views/NotesPage.vue (hunk 2 above)
Comment (SanjithVenkatesh) [reply src/views/NotesPage.vue]: Fixed

File: src/views/NotesPage.vue (hunk 1 above)
Comment (shifuchen98) [reply src/views/NotesPage.vue]: Seems `name` is sufficient for the tracking purpose and it's safe to remove the `code` property.

**Pull request merged**